config/static.c

```diff
@@ -26,103 +35,121 @@
 #include "gatekeeper_lls.h"
 #include "gatekeeper_rt.h"
 
+#define RTE_LOGTYPE_CONFIG RTE_LOGTYPE_USER2
+
+/* TODO Get the install-path via Makefile. */
+#define LUA_BASE_DIR "./lua/"
+#define LUA_CONFIG_FILE     "gatekeeper_config.lua"
+
+struct gatekeeper_config {
+	struct bp_config 	*bp_conf;
+	struct catcher_config 	*catcher_conf;
+	struct dynamic_config 	*dy_conf;
+	struct cps_config 	*cps_conf;
+	struct ggu_config 	*ggu_conf;
+	struct gk_config 	*gk_conf;
+	struct gt_config 	*gt_conf;
+	struct rt_config 	*rt_conf;
+};
+
+static lua_State *lua_state;
+
+/*
+ * The configuration data structure will be created and 
+ * filled up by its corresponding Lua configuration script.
+ */
+static struct gatekeeper_config gatekeeper_conf;
+
+static int
+set_lua_path(lua_State *l, const char *path)
+{
+	int ret;
+	char new_path[1024];
+	memset(new_path, 0, sizeof(new_path));
+
+	lua_getglobal(l, "package");
+	lua_getfield(l, -1, "path");
+
+	ret = snprintf(new_path, sizeof(new_path), "%s;%s?.lua", lua_tostring(l, -1), path);
+	RTE_ASSERT(ret < sizeof(new_path));
+
+	lua_pop(l, 1);
+	lua_pushstring(l, new_path);
+	lua_setfield(l, -2, "path");
+	lua_pop(l, 1);
+
+	return ret;
+}
+
 int
 config_and_launch(void)
 {
-	struct bp_config bp_conf;
-	struct catcher_config catcher_conf;
-	struct dynamic_config dy_conf;
-	struct cps_config cps_conf;
-	struct ggu_config ggu_conf;
-	struct gk_config gk_conf;
-	struct gt_config gt_conf;
-	struct rt_config rt_conf;
 	int ret;
+	char lua_entry_path[128];
 
-	/*
-	 * TODO Read in configuration file(s) using Lua and create
-	 * config structs for each functional block. Then launch
-	 * all functional blocks (with their config struct) that
-	 * are needed with their own lcore(s).
-	 */
+	ret = snprintf(lua_entry_path, sizeof(lua_entry_path), \
+			"%s%s", LUA_BASE_DIR, LUA_CONFIG_FILE);
+	RTE_ASSERT(ret < sizeof(lua_entry_path));
 
-	/*
-	 * TODO Decide which lcore*s* will be assigned to BP and decide
-	 * what other configuration information should be passed to
-	 * this functional block.
-	 */
-	ret = run_bp(&bp_conf);
-	if (ret < 0)
-		return ret;
+	lua_state = luaL_newstate();
```

Comment:
Shouldn't you test if this function was successful? What about the functions below?